src/components/Admin/Admin.js

```
@@ -0,0 +1,112 @@
+import '@reshuffle/code-transform/macro';
+import React, { useState, useEffect } from 'react';
+import { Link } from 'react-router-dom';
+
+import Form from 'react-bootstrap/Form';
+import Button from 'react-bootstrap/Button';
+import Container from 'react-bootstrap/Container';
+import Row from 'react-bootstrap/Row';
+import Col from 'react-bootstrap/Col';
+
+import { addNewUrl, deleteLinkById, getLinks } from '../../../backend/backend';
+import './admin.css';
+
+export default function Admin() {
+  const [inputValue, setInputValue] = useState('');
+  const [linksList, setLinksList] = useState([]);
+
+  useEffect(() => {
+    async function fetchFromDb() {
+      let links = await getLinks();
+      setLinksList([...links]);
+    }
+    try {
+      fetchFromDb();
+    } catch {
+      console.error('An error on fetch');
+    }
+  }, []);
+
+  const handleAddLink = async () => {
+    try {
+      let text = inputValue;
+
+      // prevent empty string to add in list
+      if (!text) return;
+
+      // each url will have it's own id which will be used to delete this url in the future
+      const list = await addNewUrl({
+        id: generateRandomId(),
+        url: text,
+      });
+
+      // update page with the new url
+      setLinksList(list.reverse());
+      setInputValue('');
+    } catch (error) {}
```

It's pretty dangerous to just swallow the error.

---

Do you mean because the block is empty? 

I added code to that block, let me know if this what you mean